src/sentry/integrations/github/client.py

```diff
@@ -70,99 +70,99 @@ def get_repo(self, repo: str) -> JSONData:
+                    # repositories is a single GH API network request, thus,
+                    # being acceptable to sometimes not having everything cached
+                    repo_files = filter_source_code_files(files=repo_files)
+                cache.set(key, repo_files, cache_seconds)
+            except ApiError as e:
+                json_data: JSONData = e.json
+                msg: str = json_data.get("message")
+                # TODO: Add condition for  getsentry/DataForThePeople
+                # e.g. getsentry/nextjs-sentry-example
+                if msg == "Git Repository is empty.":
+                    logger.warning(f"{repo_full_name} is empty.")
+                elif msg == "Not Found":
+                    logger.warning(f"The Github App does not have access to {repo_full_name}.")
+                else:
+                    logger.exception("An unknown error has ocurred.")
 
         return repo_files
 
     def get_trees_for_org(
         self, cache_key: str, gh_org: str, cache_seconds: int = 3600 * 24
     ) -> Dict[str, RepoTree]:
         """
-        This fetches tree representations of all repos for an org.
+        This fetches tree representations of all repos for an org and saves its
+        contents into the cache.
         """
         trees: Dict[str, RepoTree] = {}
         cache_key = f"githubtrees:repositories:{cache_key}:{gh_org}"
-        repo_key = "githubtrees:repo"
-        cached_repositories = cache.get(cache_key, [])
-        if not cached_repositories:
+        repositories = cache.get(cache_key)
+        if not repositories:
             # Simply removing unnecessary fields from the response
             repositories = [
                 {"full_name": repo["full_name"], "default_branch": repo["default_branch"]}
                 for repo in self.get_repositories(fetch_max_pages=True)
             ]
-            # XXX: In order to speed up this function we will need to parallelize this
-            # Use ThreadPoolExecutor; see src/sentry/utils/snuba.py#L358
-            for repo_info in repositories:
-                try:
-                    full_name: str = repo_info["full_name"]
-                    branch = repo_info["default_branch"]
-                    files = self.get_repo_files(full_name, branch)
-                    repo = Repo(full_name, branch)
-                    trees[full_name] = RepoTree(repo, files)
-                    cache.set(f"{repo_key}:{full_name}", trees[full_name], cache_seconds)
-                except Exception:
-                    # Catching the exception ensures that we can make progress with the rest
-                    # of the repositories
-                    logger.exception(f"We have failed to fetch the tree for {repo_info}.")
             cache.set(cache_key, repositories, cache_seconds)
             next_time = datetime.now() + timedelta(seconds=cache_seconds)
             logger.info(f"Caching trees for {gh_org} org until {next_time}.")
-        else:
```

Comment:
This else branches is fully dropped. The code did not need to be branched code.

The split view may help understand that half the code was dropped by making some adjustments:
<img width="1373" alt="image" src="https://user-images.githubusercontent.com/44410/204309870-6f8f65c2-b780-4c35-a787-8716bdde77c1.png">